File: javascript/clock/clock.js
const DAY = 24 * 60;

export class Clock {
    constructor(hh, mm=0) {
        this.minutes = (hh*60 + mm) % DAY;
        if (this.minutes < 0) {
            this.minutes += DAY;
        }
    }

    toString() {
        return [this.minutes/60, this.minutes%60]
            .map(Math.floor)
            .map(String)
            .map(val => val.padStart(2, '0'))
            .join(':');
    }

    plus(mm) {
        return new Clock(0, this.minutes + mm);
    }

    minus(mm) {
        return new Clock(0, this.minutes - mm);
    }

    equals(other) {
        return this.minutes === other.minutes;
    }
}

File: python/clock/clock.py
DAY = 24 * 60

class Clock:
    def __init__(self, hour, minute):
        self.minutes = ((((hour * 60) + minute) % DAY) + DAY) % DAY

    def __repr__(self):
        h = self.minutes // 60
        m = self.minutes % 60
        return f"{h:02}:{m:02}"

    def __eq__(self, other):
        return self.minutes == other.minutes

    def __add__(self, minutes):
        return Clock(0, self.minutes + minutes)

    def __sub__(self, minutes):
        return Clock(0, self.minutes - minutes)
